lib/ecto/query/planner.ex

```diff
@@ -1956,10 +1957,10 @@ defmodule Ecto.Query.Planner do
   end
 
   defp error!(query, expr, message) do
-    raise Ecto.QueryError, message: message, query: query, file: expr.file, line: expr.line
+    raise Ecto.QueryError, message: message, query: query, file: Map.get(expr, :file), line: Map.get(expr, :line)
   end
 
   defp error!(query, expr, message, hint) do
-    raise Ecto.QueryError, message: message, query: query, file: expr.file, line: expr.line, hint: hint
+    raise Ecto.QueryError, message: message, query: query, file: Map.get(expr, :file), line: Map.get(expr, :line), hint: hint
```

Comment:
Should we add `file` and `line` fields to `From`?

Reply:
Good idea, that would be a lot nicer than this. Will work on it.

Reply:
I made this change but right now the file/line are only being set in the `from` macro. All the other macros that call `to_query` (`select`, `filter`, etc...) won't set the file/line. I was wondering if you mind me doing that in a separate PR because it's a bit involved and would be nice to tackle it separately.

Then actually there are 2 other things I want to tackle in another PR too:

1. A nice error message when trying to preload from a fragment source.
2. A nice error message when trying to use a fragment source in a subquery. This fails because subqueries have to know which fields are in the source.